core/src/test/java/org/mule/util/TestsLogConfigurationHelper.java

```diff
@@ -63,14 +64,17 @@ public static void clearLoggingConfig()
 
     private static String findLogConfigurationPath(Class<?> testClass)
     {
-        File folder = new File(testClass.getClassLoader().getResource("").getPath().toString());
-        if (folder != null && "target".equals(folder.getParentFile().getName()))
+        String folderPath = testClass.getClassLoader().getResource("").getPath().toString();
+        folderPath = URLDecoder.decode(folderPath);
+        File folder = new File(folderPath);
+
+        if (folder != null && "target".equals(folder.getParentFile().getName()) && folder.isDirectory())
         {
             folder = folder.getParentFile();
         }
 
         File logConfigFile = FileUtils.findFileByName(folder, "log4j2-test.xml", true);
-        if (logConfigFile == null)
+        if (logConfigFile == null && logConfigFile.isFile())
```

Comment:
logConfigFile should be tested not to be a directory when trying to set the logConfigPath variable. If logConfigFile is a directory, then you assign NULL_CONFIG_FILE to it